lib/services/local/users_test.go

```diff
@@ -765,6 +768,27 @@ func TestIdentityService_GlobalWebauthnSessionDataCRUD(t *testing.T) {
 	}
 }
 
+func TestIdentityService_WebauthnSessionDataExpiry(t *testing.T) {
+	t.Parallel()
+	clock := clockwork.NewFakeClock()
+	identity := newIdentityService(t, clock)
+
+	ctx := context.Background()
+	sessionData := &wanpb.SessionData{Challenge: []byte("challenge"), UserId: []byte("userid")}
+	err := identity.UpsertWebauthnSessionData(ctx, "user", "login", sessionData)
+	require.NoError(t, err)
+
+	got, err := identity.GetWebauthnSessionData(ctx, "user", "login")
+	require.NoError(t, err)
+	require.Equal(t, sessionData, got)
+
+	// If the session data is expired but not automatically removed from
+	// the backend, it should be manually deleted on retrieval attempt.
+	clock.Advance(10 * time.Minute)
+	_, err = identity.GetWebauthnSessionData(ctx, "user", "login")
```

Comment:
If we were to Get a second time, what error would we get? Could we emulate that same error so the behavior is consistent?

Reply:
The second Get would return a NotFound error, I can add it to the test. The "webauthn session expired" error is meant to be a transient error, do you think that will cause issues?

Reply:
I think it's worth keeping the behavior consistent. The client doesn't need to know how the session expired, only that it's not there anymore.